core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -34,10 +38,28 @@ oppia.factory('UrlService', ['$window', function($window) {
       return urlParts[1] === 'embed';
     },
     getPathname: function() {
-      return $window.location.pathname;
+      return this.getCurrentUrl().pathname;
+    },
+    getParamValuesAsList: function(queryTag) {
+      var url = this.getCurrentUrl().href.slice(
```

Comment:
The name is misleading: url sounds like a string but it's really an Array of parts of a url. Maybe call it queryUrlParams or something like that. Naming is important, always take the time to choose names that are really clear since code gets read much more often than it's written.

Reply:
Done, ok, will try to use better names in the future.